src/java/org/apache/cassandra/utils/EstimatedHistogram.java

```diff
@@ -390,7 +407,15 @@ public EstimatedHistogram deserialize(DataInputPlus in) throws IOException
                 offsets[i == 0 ? 0 : i - 1] = in.readLong();
                 buckets[i] = in.readLong();
             }
-            return new EstimatedHistogram(offsets, buckets);
+            EstimatedHistogram histogram = new EstimatedHistogram(offsets, buckets);
+
+            if (histogram.isOverflowed())
+            {
+                logger.warn("Deserialized a histogram with {} values greater than the maximum of {}.",
+                            histogram.overflowCount(), histogram.getLargestBucketOffset());
```

Comment:
I think we should have at least some context for what SSTable a reader is actually being built for to go along with this message.

Reply:
the thread name would give a decent amount of context, but you could dump whole stack trace with message too I suppose but I think its sufficient like this